py/desitarget/io.py

```diff
@@ -1772,6 +1789,106 @@ def load_pixweight_recarray(inmapfile, nside, pixmap=None):
     return outdata
 
 
+def get_checksums(infiles, verbose=False, check_existing=True):
+    """Get the sha256 checksums for a list of files.
+
+    Parameters
+    ----------
+    infiles : :class:`list` or `str`
+        The full paths to a file or files.
+    verbose : :class:`bool`, optional, defaults to ``False``
+        If ``True`` then log progress and times.
+    check_existing : :class:`bool`, optional, defaults to ``True``
+        If ``True`` check if any of the `infiles` is in a directory in
+        which a .sha256sum file exists, and, if so, check generated
+        checksums for each file against the corresponding entry in the
+        relevant .sha256sum file (or *files* for `infiles` that span
+        multiple directories). An exception is raised for a mismatch.
+
+    Returns
+    -------
+    :class:`~numpy.ndarray`
+        A recarray with two columns "FILENAME" and "SHA256".
+    """
+    from subprocess import Popen, PIPE, STDOUT
+    t0 = time()
+    # ADM in case a single string was passed.
+    infiles = np.atleast_1d(infiles)
+
+    # ADM we'll first populate a dictionary with the checksums.
+    shadict = {}
+    nf = len(infiles)
+    # ADM if verbose is True, write out info for 20 blocks of files.
+    block = nf // 20 if nf // 20 else 1
+    for ifn, infile in enumerate(infiles):
+        p = Popen(['sha256sum', infile], stdout=PIPE, stderr=PIPE)
```

Comment:
My only concern is that you are using the external checksum command instead of Python's built-in `hashlib`, https://docs.python.org/3/library/hashlib.html.  Is there a performance reason for that?  This also assumes that `sha256sum` is always in `PATH`, which might cause problems testing on MacOS, where `sha256sum` isn't installed by default.

Reply:
I had no particular reason for using the external command, other than that it hadn't occurred to me that `sha256sum` might not be available on Mac OS X. Good point.

I'll happily switch, although this PR will have to go on the back-burner until after I've finished targeting for recommissioning.

Reply:
I added a function called `get_sha256sum()` that uses Python's built-in `hashlib` to generate the checksums instead of relying on the OS.